server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -776,23 +774,18 @@ private void maybeSyncGlobalCheckpoints() {
                         continue;
                     case STARTED:
                         try {
-                            shard.acquirePrimaryOperationPermit(
-                                    ActionListener.wrap(
-                                            releasable -> {
-                                                try (Releasable ignored = releasable) {
-                                                        shard.maybeSyncGlobalCheckpoint("background");
-                                                }
-                                            },
-                                            e -> {
-                                                if (!(e instanceof AlreadyClosedException || e instanceof IndexShardClosedException)) {
-                                                    logger.info(
-                                                            new ParameterizedMessage(
-                                                                    "{} failed to execute background global checkpoint sync",
-                                                                    shard.shardId()),
-                                                            e);
-                                                }
-                                            }),
-                                    ThreadPool.Names.SAME, "background global checkpoint sync");
+                            shard.runUnderPrimaryPermit(
+                                    () -> shard.maybeSyncGlobalCheckpoint("background"),
+                                    e -> {
+                                        if (e instanceof AlreadyClosedException == false
+                                                && e instanceof IndexShardClosedException == false) {
+                                            logger.warn(
```

Comment:
I did change this in this PR to from info to warn. I think warn is more appropriate.